gyroScatterEff.cpp

```diff
@@ -379,8 +516,61 @@ int main(int argc, char** argv) {
 			     numRings, numPtsPerRing,
 			     owners_d, std::string("MeshFields"));
      }
+  }
+  else if(runMode==4){ //kokkos view
+    fprintf(stderr, "mode: kokkosView\n");
+    /*  Create 2 kokkos views as eff_major and eff_minor
+     *  pass into gyroScatterKokkos
+     *  +---------------------------------------------+
+     *  |  Vector01 |
+     *  +---------------+-------------+---------------+
+     *  | vtx01 | vtx02 | vtx03 | vtx04 | vtx05 | vtx06 | ...
+     *  +-------+-------+-------+-------+-------+-------+
+     *  |c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c|c| ...
+     *  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     *   ^ ^
+     *   | |__________
+     *   |            |
+     *  +-----------------------------+
+     *  |Component|Component|Component| ...
+     *  +---------+---------+---------+
+     *  | r0 | r1 | r0 | r1 | r0 | r1 | ...
+     *  +----+----+----+----+----+----+
+     *  
+     *
+     *
+     *  -- fit as many 'extents' inside the vector size.
+     *  -- generate appropriate ranges for vector loop variables.
+     *
+     *  cabana: eff_major.access(s,a,i*gnrp1+ring)
+     *  -> eff_major[ (stride * s) + a + (VectorLength * (i*gnrp1+ring)) ];
+     *
+     *  we know that vtx = s*VectorLength+a
+     *  and vtx must vary between [0,numVerts)
```

Comment:
we can probably remove the comments from `cabana: eff_major....` to `and vtx must vary`